File: csharp/rotational-cipher/RotationalCipher.cs
using System;
using System.Linq;
using System.Text;

public static class RotationalCipher
{
    public static string Rotate(string text, int shiftKey)
        => string.Join("", text.Select(chr => chr switch
        {
            <= 'z' and >= 'a' => (char)('a' + ((chr - 'a' + shiftKey) % ('z' - 'a' + 1))),
            <= 'Z' and >= 'A' => (char)('A' + ((chr - 'A' + shiftKey) % ('Z' - 'A' + 1))),
            _ => chr
        })).ToString();
}

File: java/rotational-cipher/src/main/java/RotationalCipher.java
class RotationalCipher {
  final int shiftKey;

  RotationalCipher(int shiftKey) {
    this.shiftKey = shiftKey;
  }

  String rotate(String data) {
    StringBuilder sb = new StringBuilder(data.length());
    data.chars()
        .map(
            cp ->
                switch (Character.getType(cp)) {
                  case Character.UPPERCASE_LETTER -> shift(cp, 'A');
                  case Character.LOWERCASE_LETTER -> shift(cp, 'a');
                  default -> cp;
                })
        .forEach(sb::appendCodePoint);

    return sb.toString();
  }

  private int shift(int letter, char start) {
    return start + (letter - start + shiftKey) % 26;
  }
}
